pkg/k8s/informers/informers.go

```diff
@@ -0,0 +1,167 @@
+package informers
+
+import (
+	"github.com/rs/zerolog/log"
+	smiTrafficAccessClient "github.com/servicemeshinterface/smi-sdk-go/pkg/gen/client/access/clientset/versioned"
+	smiTrafficSpecClient "github.com/servicemeshinterface/smi-sdk-go/pkg/gen/client/specs/clientset/versioned"
+	smiTrafficSplitClient "github.com/servicemeshinterface/smi-sdk-go/pkg/gen/client/split/clientset/versioned"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+
+	configClientset "github.com/openservicemesh/osm/pkg/gen/client/config/clientset/versioned"
+	policyClientset "github.com/openservicemesh/osm/pkg/gen/client/policy/clientset/versioned"
+)
+
+type InformerCollectionOpt func(*InformerCollection)
+
+func NewInformerCollection(meshName string, stop <-chan struct{}, kubeClient kubernetes.Interface, smiTrafficSplitClient smiTrafficSplitClient.Interface,
```

Comment:
A final thought, is if we are able to properly nest clients, this could simply take a kubeconfig object, and call the NewConfigOrDie on each set of API's itself, instead of taking a forever growing list of api clients.

Reply:
Hmm that would be easier for sure. It does get weirder if someone wants to subselect informers within a particular client though

Reply:
Good point; I feel like that's probably a later iteration of this work; this set of PRs won't collapse all the various clients; just provide a shared informer collection for them to use